test/recipes/05-test_pbe.t

```diff
@@ -0,0 +1,27 @@
+#! /usr/bin/env perl
+# Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+use warnings;
+
+use OpenSSL::Test::Simple;
+use OpenSSL::Test;
+use OpenSSL::Test::Utils;
+use OpenSSL::Test qw/:DEFAULT srctop_file bldtop_dir/;
+use Cwd qw(abs_path);
+
+setup("test_pbe");
+
+plan skip_all => "PKCS5 PBE only available in legacy provider"
+    if disabled("legacy");
+
+$ENV{OPENSSL_MODULES} = abs_path(bldtop_dir("providers"));
+$ENV{OPENSSL_CONF} = abs_path(srctop_file("test", "default-and-legacy.cnf"));
+
+simple_test("test_pbe", "pbetest");
```

Comment:
Better as: `ok(run(test(["pbetest"])), "running PBE test");` and not using Test::Simple and the $ENV{OPENSSL_MODULES} setting.

Reply:
Does this include the legacy provider or is there another step needed?